docs/introduction-bdd.md

```diff
@@ -0,0 +1,97 @@
+     
+   Coming to reality Customer is a person who is an expert in dealing the things other than Technology. They need not to be aware of Edge case scenarios,beta testing and Alpha testing and so on. Even if he(customer) is familiar of the Technology stack,the only thing that is evolving in much faster pace is business, which change at constant pace.
+     
+   In this modern Agile world, teams understands the business change and structure their approach according to the business needs.Rather attempt to conceive, build, and deliver the perfect application in one fell swoop, they steadily deliver work in small and usable chunks for every few weeks.  The customer is encouraged to think in terms of what functionality will be delivered at the maximum benefit to the business and prioritize the pipeline of work accordingly.
+     
+  ![misconception between the team](images/problems.png)
+
+   This is where BDD comes in.  It bakes quality into the process by smoothing communication between stakeholders.  
+  
+  The BDD process looks like this:
+
+  A Subject-Matter-Expert (typically a business user) works with a Business Analyst to identify a business requirement. 
+  
+  *This is expressed as a story using the following template:*
+
+|As a Role
+|-
+|I request a Feature
+|To gain a Benefit
+|The speaker, who holds the Role, is the person who will gain the Benefit from the requested Feature.
+
+This can also be paraphrased variously as ...
+
+I want to achieve a specific Goal, and as a Role I should be able to accomplish this by performing Functionality.
+A Role invokes Feature to cause a Benefit[1]
+
+Software teams work best when the developers and business stakeholders
+are communicating clearly with one another. A great way to do that is to
+collaboratively specify the work that’s about to be done using automated
+acceptance tests.
+
+When the acceptance tests are written as examples, they stimulate people’s
+imaginations and help them see other scenarios they hadn’t previously
+considered.
+
+When the team write their acceptance tests collaboratively, they can develop
+their own ubiquitous language for talking about their problem domain. This
+helps them avoid misunderstandings.[2]
+   
+   
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- DAN North
+
+
+# BDD with Cucumber:
+     
+   Cucumber is a command-line tool. When you run it, it reads in your specifications from plain-language text files called features, examines them for scenarios to test, and runs the scenarios against your system. Each scenario is a list of steps for Cucumber to work through. So that Cucumber can understand these feature files, they must follow some basic syntax rules. The name for this set of rules is Gherkin.
```

Comment:
Technical jargon like "command-line" and "text files" makes this hard to digest for non-technical readers.

Add a link to the Gherkin page.